### DISCUSSION TITLE
Etherpad Basic Session

### DISCUSSION THREAD

**Discussion by jhalcomb** [ether/etherpad-lite#5579] [Q&A]
I am trying to understand the overall concept of using sessions. I can't get them working they way I thought it should work.
All I want to do is enable the 'requireSession' setting and create a session for each user and give them access to a pad.

**Here is what I tried**
First I made the "requireSession" to true in settings.json
Then I created a group (createGroup)
Then I created a pad in that group (createPadGroup)
Then I created an author for each user (createAuthor)
Then I created a session which links the author to the group. (createSession)
Then I set the session onto the users cookie under the key  "sessionID" (I am using Flask)

However, the user gets the "you do not have permission" page. They DO have the session in the cookie. I would have though if the user has a session to the group then they could access the pad that is in that group. Is this overall idea right? I couldn't find a clear explanation of how this works exactly in the docs.

**Comment by webzwo0i** [comment]
I think it should work like you described. cf https://github.com/ether/etherpad-lite/blob/develop/src/tests/backend/specs/api/sessionsAndGroups.js
I only noticed a small typo: `createPadGroup` should be `createGroupPad`

Please verify that it works with something like:
`wget --header="Cookie: sessionID=s.your-sessionID" "http://localhost:9001/p/g.the-group-pad-random-id\$some-id/export/txt"`
